src/lib/transport.js

```diff
@@ -0,0 +1,290 @@
+'use strict'
+
+const Packet = require('../utils/packet')
+const moment = require('moment')
+const cryptoHelper = require('../utils/crypto')
+const cc = require('../utils/condition')
+const co = require('co')
+const debug = require('debug')('ilp:transport')
+const assert = require('assert')
+const base64url = require('../utils/base64url')
+const BigNumber = require('bignumber.js')
+const { safeConnect } = require('../utils')
+const { createDetails, parseDetails } = require('../utils/details')
+
+function _safeDecrypt (data, secret) {
+  if (!data) return {}
+  try {
+    return parseDetails({ details: data, secret })
+  } catch (err) {
+    debug('decryption error="' + err.message + '"', 'data="' + data + '"')
+    return undefined
+  }
+}
+
+function createPacketAndCondition ({
+  id,
+  destinationAmount,
+  destinationAccount,
+  secret,
+  data,
+  headers,
+  unsafeHeaders,
+  expiresAt
+}, protocol) {
+  assert(typeof destinationAmount === 'string', 'destinationAmount must be a string')
+  assert(typeof destinationAccount === 'string', 'destinationAccount must be a string')
+  assert(Buffer.isBuffer(secret), 'secret must be a buffer')
+
+  const receiverId = base64url(cryptoHelper.getReceiverId(secret))
+  const address = destinationAccount + '.~' + protocol + '.' + receiverId +
+    (id ? ('.' + id) : '')
+
+  const details = createDetails({
+    unsafeHeaders: Object.assign({}, unsafeHeaders),
+    headers: Object.assign({
+      'Expires-At': expiresAt
+    }, headers),
+
+    data: data,
+    secret: secret
+  })
+
+  const packet = Packet.serialize({
+    account: address,
+    amount: destinationAmount,
+    data: details
+  })
+
+  const condition = base64url(cc.toCondition(
+    cryptoHelper.hmacJsonForPskCondition(
```

Comment:
This isn't right. Why is it using the hmacJson function? This should hmac the packet buffer. The hmacJson method should not be used anymore and should be removed (along with the `canonical-json` dependency)

This is also why it's important to have a test to match the output against a specific output. The current tests would not catch this because the sender and recipient are consistent with one another but they don't match the spec.